app/controllers/root_controller.rb

```diff
@@ -164,7 +165,7 @@ def block_empty_format
 
   def prepare_publication_and_environment
     publication = publication_with_places(
-      params[:postcode], params[:slug], params[:edition]
+      PostcodeSanitizer.sanitize(params[:postcode]), params[:slug], params[:edition]
     )
```

Comment:
Done. Now we only call `PostcodeSanitizer` once as the `@postcode` assignment is above the `prepare_publication_and_environment` call, and that method now uses `@postcode` (and it's still OK if it's nil).